core/query/sql.ml

```diff
@@ -173,16 +177,25 @@ class virtual printer =
     let pp_from_clause ppf fc =
       match fc with
         | TableRef (t, x) -> Format.fprintf ppf "%a as %s" self#pp_quote t (string_of_table_var x)
-        | Subquery (q, x) -> Format.fprintf ppf "(%a) as %s" pr_q q (string_of_table_var x) in
+        | Subquery (false, q, x) -> Format.fprintf ppf "(%a) as %s" pr_q q (string_of_table_var x)
+        | Subquery (true, q, x) -> Format.fprintf ppf "lateral (%a) as %s" pr_q q (string_of_table_var x) in
     let pp_where ppf condition =
       match condition with
         | Constant (Constant.Bool true) -> ()
-        | _ -> Format.fprintf ppf "\nwhere %a" pp_os_condition condition in
-    Format.fprintf ppf "select %a\nfrom %a%a%a"
-      self#pp_fields fields
-      (self#pp_comma_separated pp_from_clause) tables
-      pp_where condition
-      pp_orderby os
+        | _ -> Format.fprintf ppf "\nwhere %a" pp_os_condition condition
+      in
+      if fDistinct then
+        Format.fprintf ppf "select distinct %a\nfrom %a%a%a"
+        self#pp_fields fields
+        (self#pp_comma_separated pp_from_clause) tables
+        pp_where condition
+        pp_orderby os
+      else
+        Format.fprintf ppf "select %a\nfrom %a%a%a"
```

Comment:
Ok. I'd still prefer to generate "select" and "union" rather than "select all" and "union distinct", so the code is specialized in such a way that the unnecessary specification is omitted.